base/string.jl

```diff
@@ -1007,38 +1011,71 @@ function indentation(s::AbstractString)
     count, true
 end
 
-function unindent(s::AbstractString, indent::Int)
-    indent == 0 && return s
-    buf = IOBuffer(Array(UInt8,endof(s)), true, true)
+"
+Removes leading indentation from string
+
+### Returns:
+* `ASCIIString` or `UTF8String` of multiline string, with leading indentation of `indent` removed
+"
+function unindent(str::AbstractString, indent::Int)
+    indent == 0 && return str
+    pos = start(str)
+    endpos = endof(str)
+    # Note: this loses the type of the original string
+    buf = IOBuffer(Array(UInt8,endpos), true, true)
     truncate(buf,0)
-    a = i = start(s)
     cutting = false
-    cut = 0
-    while !done(s,i)
-        c,i_ = next(s,i)
-        if cutting && (c == ' ' || c == '\t')
-            a = i_
-            cut += blank_width(c)
-            if cut == indent
-                cutting = false
-            elseif cut > indent
+    col = 0     # current column (0 based)
+    tabwid = tab_indentation_width
+    while pos <= endpos
+        ch, pos = next(str,pos)
+        if cutting
+            if ch == ' '
+                col += 1
+            elseif ch == '\t'
+                col = div(col + tabwid, tabwid) * tabwid
+            elseif ch == '\n'
+                # Now we need to output enough indentation
+                for i = 1:max(0, col-indent) ; write(buf, ' ') ; end
```

Comment:
I was just following the style of the code I was replacing, but I can make those 3 lines.